pkg/storage/snapshot/snapshot.go

```diff
@@ -195,10 +202,14 @@ func (ctrl *VMSnapshotController) updateVMSnapshot(vmSnapshot *snapshotv1.Virtua
 						}
 					}
 				}
-			} else if canUnlockSource(vmSnapshot, content) {
-				if _, err := source.Unlock(); err != nil {
-					return 0, err
+				canRemoveFinalizer = false
+			} else {
+				if canUnlockSource(vmSnapshot, content) {
```

Comment:
nit: you can use `if else` and we avoid one level nesting

Reply:
Then I'd have to have duplicate [1] for `else if` and `else` case. I think once is better:

[1] https://github.com/mhenriks/kubevirt/blob/snapshot-api-status-subresource/pkg/storage/snapshot/snapshot.go#L214